mirgecom/viscous.py

```diff
@@ -275,63 +277,170 @@ def viscous_flux(state, grad_cv, grad_t):
+def viscous_divergence_flux(discr, state_pair, grad_cv_pair, grad_t_pair,
+                         beta=0., gamma=0, **kwargs):
+    r"""Return a viscous facial flux for the divergence operator.
+
+    The flux is defined as:
+
+    .. math::
+
+        f_{\text{face}} = \frac{1}{2}\left(\mathbf{f}_v^+
+        + \mathbf{f}_v^-\right)\cdot\hat{\mathbf{n}}
+        + \frac{\beta}{2}\left(\mathbf{f}_v^+ - \mathbf{f}_v^-\right),
+
+    with viscous fluxes ($\mathbf{f}_v$), and the outward pointing
+    face normal ($\hat{\mathbf{n}}$).
 
     Parameters
     ----------
     discr: :class:`~grudge.eager.EagerDGDiscretization`
 
         The discretization to use
 
-    state_tpair: :class:`~grudge.trace_pair.TracePair`
+    state_pair: :class:`~grudge.trace_pair.TracePair`
 
         Trace pair of :class:`~mirgecom.gas_model.FluidState` with the full fluid
         conserved and thermal state on the faces
 
-    grad_cv_tpair: :class:`~grudge.trace_pair.TracePair`
+    grad_cv_pair: :class:`~grudge.trace_pair.TracePair`
 
         Trace pair of :class:`~mirgecom.fluid.ConservedVars` with the gradient of the
         fluid solution on the faces
 
-    grad_t_tpair: :class:`~grudge.trace_pair.TracePair`
+    grad_t_pair: :class:`~grudge.trace_pair.TracePair`
 
         Trace pair of temperature gradient on the faces.
 
-    local: bool
-
-        Indicates whether to skip projection of fluxes to "all_faces" or not. If
-        set to *False* (the default), the returned fluxes are projected to
-        "all_faces".  If set to *True*, the returned fluxes are not projected to
-        "all_faces"; remaining instead on the boundary restriction.
-
     Returns
     -------
     :class:`~mirgecom.fluid.ConservedVars`
 
         The viscous transport flux in the face-normal direction on "all_faces" or
         local to the sub-discretization depending on *local* input parameter
     """
-    actx = state_tpair.int.array_context
-    dd = state_tpair.dd
-    dd_all_faces = dd.with_dtag("all_faces")
+    actx = state_pair.int.array_context
+    normal = thaw(discr.normal(state_pair.dd), actx)
+
+    f_int = viscous_flux(state_pair.int, grad_cv_pair.int,
+                         grad_t_pair.int)
+    f_ext = viscous_flux(state_pair.ext, grad_cv_pair.ext,
+                         grad_t_pair.ext)
+    f_pair = TracePair(state_pair.dd, interior=f_int, exterior=f_ext)
+    q_pair = TracePair(state_pair.dd, interior=state_pair.int.cv,
+                       exterior=state_pair.ext.cv)
+    from arraycontext import outer
+    jump_term = -gamma*outer(q_pair.diff, normal)/2
+
+    return divergence_flux(trace_pair=f_pair, normal=normal,
+                           alpha=jump_term, beta=beta)
+
+
+def viscous_flux_on_element_boundary(
+        discr, gas_model, boundaries, interior_state_pairs,
+        domain_boundary_states, grad_cv, interior_grad_cv_pairs,
+        grad_t, interior_grad_t_pairs, quadrature_tag=None,
```

Comment:
```suggestion
        grad_t, interior_grad_t_pairs, discretization_tag=None,
```
(here and elsewhere)

Reply:
Bump to issue? Document to be a discretization tag?

Reply:
Made an issue, but eager to tackle it here if we can.